projects/kit/components/rating/rating.template.html

```diff
@@ -0,0 +1,26 @@
+<div
+    #parent
+    role="img"
+    class="tui-rating"
+    [class.tui-rating--touched]="touched"
+    [class.tui-rating--disable]="disabled"
+    [class.tui-rating--readonly]="tuiRateReadonly"
+    [attr.aria-label]="ratingState.selectedRate + ' Stars'"
+    [attr.tabindex]="ratingState.selectedRate === 0 && enabled ? 0 : -1"
+    (mouseleave)="ratesMouseLeave(parent)"
+>
+    <ng-container *tuiRepeatTimes="let index of tuiRateMax">
```

Comment:
Let's change the layout this way:
```
<svg *tuiRepeatTimes unfilled (mouseenter/mouseleave)>
<div absolutely positioned with pointer events none [style.width.%]="value / max * 100">
  <svg *tuiRepeatTimes filled>
</div>
```
This way we have a layer of inert stars showing the actual rating with subdivisions:

![image](https://user-images.githubusercontent.com/11832552/146794280-c4cd8d16-3081-4ddb-82d8-80ec57e733ee.png)

![image](https://user-images.githubusercontent.com/11832552/146794379-3476ecfe-25d2-4614-be80-ad36425fef45.png)

Reply:
But if I want a custom icon instead the star icon, what should I do?

Reply:
Customization of the icons stays the same, the idea is that there are two layers of icons overlaying each other so that we can show half an icon when the value is not an integer.